interview-questions.md

```diff
@@ -9,89 +9,89 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 
 
 7. What are props in React?
 
-  Your answer:
+  Your answer: props are used to define a variable of a child component. This way, the child component can call on that prop to gain access to the objects in its parent component.
 
-  Researched answer:
+  Researched answer: props is a keyword in react for passing info from the parent component to the child. 
 
 
 
 8. What is `this` in JavaScript?
 
-  Your answer:
+  Your answer: this is a keyword used to refer back to an object it belongs to.
 
-  Researched answer:
+  Researched answer: refers to an object, that object which is executing the current bit of javascript code.
 
 
 
 9. STRETCH: What is a ternary operator in JavaScript?
 
-  Your answer:
+  Your answer: ternary operator in javascript is used as a simpler way to write an if else statement, only if there are 2 possible outcomes.
 
-  Researched answer:
+  Researched answer: Ternary operators are handy shortcuts for creating if/else statements in JavaScript. Ternary operators allow JavaScript developers to make very clean and concise code.
 
 
 
 10. STRETCH: What is an anonymous function in JavaScript?
 
-  Your answer:
+  Your answer: Not too sure, but I think it is a function that doesn't take in an argument.
 
-  Researched answer:
+  Researched answer: Anonymous functions are functions without a name and is not accessible after its initial creation.
 
 
 
 ## Looking Ahead: Terms for Next Week
-- Ruby
-- Object oriented programming
-- RSpec
-- Instance variable
-- Ruby blocks
-- Ruby hashes
-- `getter` and `setter` methods in Ruby
+- Ruby: a scripting language used for building web applications.
+- Object oriented programming: a computer programming model that organized software design around data, or objects rather than funcion and logic.
+- RSpec: a computer domain specific language testing tool written in Ruby so that you can test Ruby code.
+- Instance variable: variables that are defined in a class where each object of the class has its own copy of the instance variables. They are created outside the counstructors and methods, or blocks in the class.
+- Ruby blocks: a way of grouping statements, and may appear only in the source adjacent to a method call.
+- Ruby hashes: a collection of key-value pairs. Similar to an array, except that indexing is done via arbitrary keys of any object type, not an integer index.
+- `getter` and `setter` methods in Ruby: allows developers to access a class's instance variable from outside the class. Getter methods are used to get the value of an instance variable while the setter methods are used to set the value of an instance variable of some class.
```

Comment:
excellent job on this file! very concise answers that get your point across. yay!